src/backend/BSDispatcher/Constraints.pm

```diff
@@ -114,8 +114,9 @@ sub oracle {
     return 0 if $constraints->{'hardware'}->{'physicalmemory'} && getmbsize($constraints->{'hardware'}->{'physicalmemory'}) > $memory;
     if ($constraints->{'hardware'}->{'cpu'}) {
       return 0 unless $worker->{'hardware'}->{'cpu'};
-      my %workerflags = map {$_ => 1} @{$worker->{'hardware'}->{'cpu'}->{'flag'} || []};
-      return 0 unless grep {$workerflags{$_}} @{$constraints->{'hardware'}->{'cpu'}->{'flag'} || []};
+      for my $flag (@{$worker->{'hardware'}->{'cpu'}->{'flag'} || []}) {
+	  return 0 unless grep {$flag} @{$constraints->{'hardware'}->{'cpu'}->{'flag'} || []};
+      }
```

Comment:
Bah! How about:
```
my %workerflags = map {$_ => 1} @{$worker->{'hardware'}->{'cpu'}->{'flag'} || []};
return 0 if grep {!$workerflags{$_}} @{$constraints->{'hardware'}->{'cpu'}->{'flag'} || []};
```

Reply:
And what on earth is `{$flag}` supposed to do? You didn't test this, did you? ;-)